compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -36,14 +40,32 @@ namespace iree_compiler {
 
 namespace {
 
+// Transform gpu.barrier -> amdgpu.lds_barrier
```

Comment:
nit: Add a brief description of why we're allowed to/want to do this. Per my understanding, `lds_barrier` only waits on lds ops while `gpu.barrier` waits on global memory operations as well. Because IREE only ever uses lds as it's form of shared memory (and isn't allowed to somehow allocate a slab of global memory to treat as shared, not really sure if that would ever make sense), we can always replace `gpu.barrier` with the strictly better `lds_barrier`.

Reply:
Updated